src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -672,6 +694,51 @@ Leader::UpdateStatus Leader::UpdateTlv(NetworkDataTlv &aTlv, const NetworkDataTl
     return status;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_SIGNAL_NETWORK_DATA_FULL
+
+void Leader::CheckForNetDataGettingFull(const NetworkData &aNetworkData, uint16_t aOldRloc16)
+{
+    // Determines whether there is still room in Network Data to register
```

Comment:
with a check method, some result of the check needs to be provided either a boolean or an object whose size can be verified. It's not clear from the comments how the result is delivered?

Reply:
Added new text in comment to clarify this. We use callback by invoking `Get<Notifier>().SignalNetworkDataFull()` (which then may call user callback and inform any other module that may be intrested).  

The reason went for this model (callback) is that it can be triggered from different paths in the code (e.g. local netdata getting full, etc).